examples/protyping/pymedphys-segmentation/data_generator-bf.py

```diff
@@ -0,0 +1,249 @@
+# %load generator
+import functools
+from pathlib import Path
+from random import randint
+
+import numpy as np
+
+import matplotlib.pyplot as plt
+
+import skimage.draw
+import skimage.transform
+
+# Perhaps use tf.io instead
+import pydicom
+
+import tensorflow as tf
+
+# TODO - Modulate get_item
+
+
+class data_gen(tf.keras.utils.Sequence):
+    def __init__(
+        self,
+        input_paths,
+        context_paths,
+        label_paths,
+        context,
+        batch_size,
+        structure_names,
+        resize,
+    ):
+        self.input_paths = input_paths
+        self.context_paths = context_paths
+        self.label_paths = label_paths
+        self.context = context
+        self.batch_size = batch_size
+        self.structure_names = structure_names
+        self.resize = resize
+
+        for path in self.label_paths:
+            _ = self.pre_cached_structures(path)
+
+        self.on_epoch_end()
+
+    @functools.lru_cache()
```

Comment:
But, it is CPU RAM this would be using not GPU RAM. So, depending on what RAM is running out it might be okay to leave it. You can also put a cache number in here that makes it only cache the most recent 20 calls or something like that.

You also could build your own class that when initialised it loads the DICOM structure and saves its results to disk. Then subsequent times that class is called it just loads your cached disk result.